olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -152,8 +262,48 @@ NetworkAsyncCallback GetRetryCallback(
                                         "Operation Cancelled."));
             });
       } else {
-        callback(network::HttpResponse(network::Network::ErrorCode::UnknownError,
-                                       "Unknown error."));
+        callback(network::HttpResponse(
+            network::Network::ErrorCode::UnknownError, "Unknown error."));
+      }
+    }
+  };
+}
+
+NetworkAsyncCallback GetNewRetryCallback(
```

Comment:
This should be moved to Network impl.
Retry mechanism with backoff policy should be used by everyone.
Please add ticket in backlog to move this.